nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -69,6 +71,27 @@ public class StandardOauth2AccessTokenProvider extends AbstractControllerService
         .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
         .build();
 
+    public static AllowableValue BASIC_AUTHENTICATION = new AllowableValue(
+        "basic",
+        "Basic Authentication",
+        "Send client authentication as HTTP Basic authentication"
+    );
+
+    public static AllowableValue REQUEST_BODY = new AllowableValue(
+        "request_body",
+        "Request Body",
+        "Send client authentication in request body. Note: users should prefer Basic Authentication unless authorization server does not support it."
+    );
```

Comment:
Although this works, recommend creating an `enum` that implements the `DescribedValue` interface. See other examples such as `ContentEncodingStrategy` and `AuthorizationScheme` in other modules. The `getValue()` method should return `name()`, and allows for better value comparison in other methods.